Выполнить ФЗ9 21.06

AleksandrK1986 opened the pull request.

AleksandrK1986 committed:
Выполнить ФЗ9 21.06

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,28 +1,93 @@
+        List<Film> films = filmService.getTopFilms();
+        for (int i = 0; i < count && i < films.size(); i++) {
+            topFilms.add(films.get(i));
+        }
```
зачем ты тут делаешь перебор и перекладываешь из одного списка в другой? ты же можешь спокойно работать с теми данными, которые вернул тебе сервис.

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/controller/Controller.java:
```diff
@@ -2,50 +2,33 @@
+    protected T update(T object) {
+        Validate.validateOrGetException(object);
+        updateObject(object);
+        log.info("Обновление объекта {}      Объект с ID {} обновлен успешно", LocalDateTime.now(), object.toString());
```
вся логика по созданию обновлению проверкам, должно быть перемещена в сервис. По сути в этом задании вместо абстрактного контроллера можно завести абстрактный сервис. 
Заведение абстрактного сервиса это по желанию. Не является обязательным требованием.

AleksandrK1986 committed:
refactor: поправить замечания от 22.06

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -20,16 +19,22 @@ public FilmController(FilmService filmService) {
-        return create(film);
+    public Film create(@RequestBody Film film) {
+        int idFilm = filmService.createFilm(film);
+        log.info("Создание фильма {}      Фильм с ID {} создан успешно", LocalDateTime.now(), film.toString());
```
на будущее логи тоже следует вынести в сервис

AleksandrK1986 committed:
refactor:
1) выполнить ФЗ 10 спринта
2) поправить замечание с ФЗ 9 в части переноса записи логов из контролеров в сервисы
